components/Movies.js

```diff
@@ -0,0 +1,183 @@
+import React, {useState, useEffect} from 'react';
+// importing core componenets from react native
+import {
+	View, 
+	Text, 
+	ActivityIndicator, 
+	Image, 
+	Button, 
+	TouchableOpacity, 
+	ScrollView, 
+	StatusBar,
+	SafeAreaView,
+	StyleSheet, 
+	Share
+} from 'react-native';
+import styled from 'styled-components/native';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+import { FETCH_URL } from '../utils/urls';
+
+const Movies = () => {
+
+	const [movieList, setMovieList] = useState([]);
+	const [loading, setLoading] = useState(false);
+
+	// random movie from movieList array, returns only one item
+	const randomMovie = movieList[Math.floor(Math.random() * 	movieList.length)];
+
+	useEffect(() => {
+		generateMovie();
+	}, []);
+
+	const generateMovie = () => {
+		fetch(FETCH_URL)
+			.then((res) => res.json())
+			.then((json) => {
+				setMovieList(json.results);
+				//setLoading(true)
+			})
+			.finally(() => setLoading(false))
+	};
+
+	// share function, that gives possibility to share directly from your phone and send a message with movie title 
+	const onShare = async () => {
+    try {
+      const result = await Share.share({
+        message: `Hey! :) I found a great movie for tonight! It called ${randomMovie.title}. Can you buy popcorn?`,
+      });
+      if (result.action === Share.sharedAction) {
+        if (result.activityType) {
+          // shared with activity type of result.activityType
+        } else {
+          // shared
+        }
+      } else if (result.action === Share.dismissedAction) {
+        // dismissed
+      }
```

Comment:
Interessting function! Why do you need all the if/else-statements, if there's no code inside?